streams/early_zip_readable_streams.ts

```diff
@@ -84,26 +84,22 @@
 export function earlyZipReadableStreams<T>(
   ...streams: ReadableStream<T>[]
 ): ReadableStream<T> {
-  const readers = streams.map((s) => s.getReader());
+  const readers = streams.map((stream) => stream.getReader());
   return new ReadableStream<T>({
-    async start(controller) {
-      try {
-        loop:
-        while (true) {
-          for (const reader of readers) {
-            const { value, done } = await reader.read();
-            if (!done) {
-              controller.enqueue(value!);
-            } else {
-              await Promise.all(readers.map((reader) => reader.cancel()));
-              break loop;
-            }
-          }
+    async pull(controller) {
+      for (const reader of readers) {
+        const { done, value } = await reader.read();
+        if (done) {
+          await Promise.all(
+            readers.map((reader) => reader.cancel("early_zip_ended")), // A better cancel message should maybe be put here?
```

Comment:
Is it correct to say it ended early though since if all the streams are of the same length then the first stream will be detected as finished, with all streams getting this message even though they're all empty.

Maybe something like \`Early Zip Stream ended at index: ${i}`?

Reply:
Good point. `Stream ended at index ${i}` SGTM—no need to reiterate the class name.